solutions/cses-1191.mdx

```diff
@@ -0,0 +1,136 @@
+---
+id: cses-1191
+source: CSES
+title: Cyclic Array
+author: Chongtian Ma
+---
+
+## Explanation
+
+We can turn this into a graph problem, or more precisely, a tree. Let's add an edge from $u$ to $v$ if the maximum length of the subarray that starts at $v$ has to end at $u-1$, and so the next subarray must start at $u$. Remember the array is cyclic, so $u$ is not necessarily greater than $v$. This can be found using basic two pointers.
+
+We can use binary lifting to find the start index of any next subarray in $\log(n)$ time. We can also track the number of elements between two subarrays.
+
+Finally, we can binary search over the number of subarrays for the minimum. If we can use $k$ subarrays to cover the whole array, then we must also be able to use $k+1$ subarrays.
+
+## Implementation
+
+**Time Complexity: $\mathcal O(n \log(n))$**
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+const int MAXN = 2e5 + 5;
+const int MAXL = 20; // approx maximum log of n
+
+// binary lifting arrays:
+
+/*
+ * next[i][j] stores the start point of the next
+ * 2^j'th subarray if we choose this subarray starting at i
+ */
+array<array<int, MAXL>, MAXN> nxt;
+
+/* len[i][j] stores the total amount of elements between
+ * subarray starting at i and the next 2^j'th subarray
+ */ 
+array<array<ll, MAXL>, MAXN> len;
+
+int main(){
+	ios::sync_with_stdio(false);
+	cin.tie(0);
+	
+	int n;
+	ll k;
+	scanf("%d%lld", &n, &k);
```

Comment:
why use `scanf`?
completely fine but other times i see you using `cin`

Reply:
doesn't matter lol